RecoTracker/LSTCore/src/alpaka/Quintuplet.h

```diff
@@ -141,22 +144,22 @@ namespace lst {
     inline void setData(QuintupletsBuffer& buf) { data_.setData(buf); }
   };
 
-  ALPAKA_FN_ACC ALPAKA_FN_INLINE bool checkIntervalOverlap(const float& firstMin,
-                                                           const float& firstMax,
-                                                           const float& secondMin,
-                                                           const float& secondMax) {
+  ALPAKA_FN_ACC ALPAKA_FN_INLINE bool checkIntervalOverlap(const float firstMin,
+                                                           const float firstMax,
+                                                           const float secondMin,
+                                                           const float secondMax) {
     return ((firstMin <= secondMin) && (secondMin < firstMax)) || ((secondMin < firstMin) && (firstMin < secondMax));
   };
 
   ALPAKA_FN_ACC ALPAKA_FN_INLINE void addQuintupletToMemory(struct lst::Triplets& tripletsInGPU,
                                                             struct lst::Quintuplets& quintupletsInGPU,
-                                                            unsigned int innerTripletIndex,
-                                                            unsigned int outerTripletIndex,
-                                                            uint16_t& lowerModule1,
-                                                            uint16_t& lowerModule2,
-                                                            uint16_t& lowerModule3,
-                                                            uint16_t& lowerModule4,
-                                                            uint16_t& lowerModule5,
+                                                            const unsigned int innerTripletIndex,
+                                                            const unsigned int outerTripletIndex,
+                                                            const uint16_t lowerModule1,
+                                                            const uint16_t lowerModule2,
+                                                            const uint16_t lowerModule3,
+                                                            const uint16_t lowerModule4,
+                                                            const uint16_t lowerModule5,
                                                             float& innerRadius,
                                                             float& bridgeRadius,
                                                             float& outerRadius,
```

Comment:
shouldn't these be by value as well (or const ref)?

Reply:
Possibly. However, tracing this down is not trivial, as they are propagated to multiple functions. I can look into that as well but it will take more time. Do we want to do it? If so, this PR or another?